app/routes/exercises.rb

```diff
@@ -66,6 +66,28 @@ class Exercises < Core
         redirect '/dashboard'
       end
 
+      post '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may request help."
+          redirect '/'
+        end
+        exercise.request_help!
+        flash[:success] = "You have requested help on this submission."
+        redirect "/submissions/%s" % exercise.submissions.last.key
+      end
+
+      delete '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may cancel a request for help."
+          redirect '/'
+        end
```

Comment:
I agree 72,73, 83 and 84 are my changes. I did not see tests for negative conditions for eg  (61,62). Yes I have not written tests for that!